src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +39,14 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+
+        return IntStream.range(0, numbers.size())
+                .mapToObj(i -> i % 2 == 1 ? numbers.get(i) - 1 : numbers.get(i))
+                .toList().stream()
+                .filter(num -> num % 2 != 0)
+                .mapToDouble(Integer::doubleValue)
+                .average()
```

Comment:
Reduce the number of map operations as much as possible.
Check other methods.

see example:
```suggestion
                .mapToDouble(i -> i % 2 == 1 ? numbers.get(i) - 1 : numbers.get(i))
                .toList().stream()
                .filter(num -> num % 2 != 0)
                .average()
```